### PR TITLE
Change @RetryableTopic properties SameIntervalTopicReuseStrategy behavior to SINGLE_TOPIC

### PR DESCRIPTION
**Pull request by Wzy19930507**
Resolves https://github.com/spring-projects/spring-kafka/issues/3052

* Change @RetryableTopic properties SameIntervalTopicReuseStrategy behavior to SINGLE_TOPIC
* Change unit test SameIntervalTopicReuseStrategy
* Add doc

### CODE REVIEW

**Comment by Wzy19930507** [comment]
@sobychacko Hi, seems like an unrelated error, please provide some help.

```
TestOOMError > testOOMCMLC() FAILED
    java.util.ConcurrentModificationException
        at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1604)
        at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
        at org.springframework.kafka.listener.TestOOMError.testOOMCMLC(TestOOMError.java:89)
```

File: spring-kafka-docs/src/main/antora/modules/ROOT/pages/retrytopic/topic-naming.adoc [hunk 1]
```diff
@@ -111,35 +111,33 @@ This "final" retry topic will be suffixed with the provided or default suffix, a
...
-* -retry-16000
-
-This will be the default in a future release.
+* -retry-16000-0
+* -retry-16000-1
+* -retry-16000-2
+* ...
+* -retry-16000-224
 
 [source, java]
```
Comment (sobychacko) [inline spring-kafka-docs/src/main/antora/modules/ROOT/pages/retrytopic/topic-naming.adoc:136]: We might want to add a short sentence stating that the following code snippet is changing the default behavior - something like `If multiple topics are required, then that can be done using the following configuration.`

File: spring-kafka-docs/src/main/antora/modules/ROOT/pages/retrytopic/how-the-pattern-works.adoc [hunk 2]
```diff
@@ -16,8 +16,8 @@ IMPORTANT: You can set the `AckMode` mode you prefer, but `RECORD` is suggested.
 IMPORTANT: At this time this functionality doesn't support class level `@KafkaListener` annotations.
 
 When using a manual `AckMode` with `asyncAcks` set to true, the `DefaultErrorHandler` must be configured with `seekAfterError` set to `false`.
-Starting with versions 2.9.10, 3.0.8, this will be set to true unconditionally for such configurations.
-With earlier versions, it was necessary to override the `RetryConfigurationSupport.configureCustomizers()` method to set the property to `true`.
+Starting with versions 2.9.10, 3.0.8, this will be set to `false` unconditionally for such configurations.
```
Comment (sobychacko) [inline spring-kafka-docs/src/main/antora/modules/ROOT/pages/retrytopic/how-the-pattern-works.adoc:19]: Was it `false` in those releases?

File: spring-kafka-docs/src/main/antora/modules/ROOT/pages/whats-new.adoc [hunk 3]
```diff
@@ -40,4 +40,9 @@ See xref:retrytopic/features.adoc#exc-based-custom-dlt-routing[Routing of messag
 === After Rollback Processing
 
 A new `AfterRollbackProcessor` API `processBatch` is provided.
-See xref:kafka/annotation-error-handling.adoc#after-rollback[After-rollback Processor] for more information.
\ No newline at end of file
+See xref:kafka/annotation-error-handling.adoc#after-rollback[After-rollback Processor] for more information.
+
+[[x32-retry-topic]]
+=== Change @RetryableTopic SameIntervalTopicReuseStrategy default value
+Change `@RetryableTopic` property SameIntervalTopicReuseStrategy default value to SINGLE_TOPIC.
```
Comment (sobychacko) [inline spring-kafka-docs/src/main/antora/modules/ROOT/pages/whats-new.adoc:47]: Code formatting needed for `SameIntervalTopicReuseStrategy` and `SINGLE_TOPIC.`

**Review by sobychacko — CHANGES_REQUESTED** [review]

File: spring-kafka-docs/src/main/antora/modules/ROOT/pages/retrytopic/how-the-pattern-works.adoc (hunk 2 above)
Comment (Wzy19930507) [reply spring-kafka-docs/src/main/antora/modules/ROOT/pages/retrytopic/how-the-pattern-works.adoc:19]: yes, When using a manual `AckMode` with `asyncAcks` set to `true`, will unconditionally set to `false`
see https://github.com/spring-projects/spring-kafka/issues/2707 and https://github.com/spring-projects/spring-kafka/issues/2702.

File: spring-kafka-docs/src/main/antora/modules/ROOT/pages/whats-new.adoc (hunk 3 above)
Comment (Wzy19930507) [reply spring-kafka-docs/src/main/antora/modules/ROOT/pages/whats-new.adoc]: thanks, fix it.

File: spring-kafka-docs/src/main/antora/modules/ROOT/pages/retrytopic/topic-naming.adoc (hunk 1 above)
Comment (Wzy19930507) [reply spring-kafka-docs/src/main/antora/modules/ROOT/pages/retrytopic/topic-naming.adoc:138]: Thanks, fix it.

**Pull request merged**